src/interceptors/request.ts

```diff
@@ -109,4 +113,53 @@ export class CacheRequestInterceptor<D>
 
     return config;
   };
+
+  private isMethodAllowed = (
+    axios: AxiosCacheInstance,
+    method: Method = 'get',
+    properties?: Partial<CacheProperties>
+  ): boolean => {
+    const requestMethod = method.toLowerCase();
+    const allowedMethods = properties?.methods || axios.defaults.cache.methods;
+
+    for (const method of allowedMethods) {
+      if (method.toLowerCase() === requestMethod) {
+        return true;
+      }
+    }
+
+    return false;
+  };
+
+  private setRequestHeaders = (
+    cache: StaleStorageValue | EmptyStorageValue,
+    config: CacheRequestConfig<D> & { cache: Partial<CacheProperties> | undefined }
+  ) => {
+    if (!config.cache) return;
+    config.headers ||= {};
+
+    const { etag, modifiedSince } = config.cache;
+
+    if (etag) {
+      const etagValue = etag === true ? cache.data?.headers[Header.ETag] : etag;
+      if (etagValue) {
+        config.headers[Header.IfNoneMatch] = etagValue;
+      }
+    }
+
+    if (
+      modifiedSince &&
+      cache.state === 'stale'
```

Comment:
I'm not sure it's useful to check for stale here, it should be like etag, and simply use it if provided.

Reply:
But i only have the createdAt property set if this is a StaleStorageValue, as the empty one is undefined.

Reply:
ha ! I've read it too fast, for `if-modified-since` header, you're supposed to send the `last-modified` header, not a local date (target server might not be in sync with world time), just like you use the `etag` header for `if-none-match`